### ISSUE TITLE
feature request: create ability to create completly costum component with variant support 

### ISSUE THREAD

**Issue by zakidze** [mantinedev/mantine#3604]
i would love to see the ability to create complelty costum component that dont exsist in mantine (create single part component and multi part component) 
example: the menu have alot of compoents inside it so you style each one and at the end expose variants so it can be adapted in diffrent senario with base styles and modifiers styles multi sizes compond variants...

but i want to build complelty costum components

**Comment by rtivital** [comment]
You can build any kind of components on your side, there are no limitations from Mantine side

**Comment by zakidze** [comment]
yes i really like mantine 1 day trying it  good luck with the development
